src/mover_policies_containers_creation_callback.erl

```diff
@@ -0,0 +1,432 @@
+           {add_acl, [{container, cookbook_artifacts}], [create, read, update, delete], [{group, users}]},
+
+           %% clients
+
+           {add_acl, [{container, policies}],           [read], [{group, clients}]},
+           {add_acl, [{container, policy_groups}],      [read], [{group, clients}]},
+           {add_acl, [{container, cookbook_artifacts}], [read], [{group, clients}]}
+          ]
+         }
+        ]).
+
+migration_init() ->
+    %% TODO: remove this when testing is complete:
+    Nuke = <<"delete from containers where name in ('policies', 'policy_groups', 'cookbook_artifacts');">>,
+    sqerl:execute(Nuke),
+
+    %% TODO: this fails if the pool is already created. Should be tolerant of
+    %% errors and also should tear this down after the migration
+    mv_oc_chef_authz_http:create_pool(),
+    %% TODO: superuser name is somewhat configurable, it's set in oc_erchef's sys.config
+    {ok, SuperuserResults} = sqerl:adhoc_select([username, id, authz_id], users, {username, equals, <<"pivotal">>}),
+    Superuser = db_results_to_requestor(hd(SuperuserResults)),
+
+    %% Should do this for each container type and uniq the orgs just in case a migration partially bombs (?)
+    FindOrgsWithNoPoliciesContainer = <<"SELECT orgs.name, orgs.id, orgs.authz_id FROM orgs "
+                                        "LEFT JOIN containers ON orgs.id = org_id AND containers.name='policies' "
+                                        "WHERE containers.name IS NULL;">>,
+    {ok, OrgsToMigrateResults} = sqerl:execute(FindOrgsWithNoPoliciesContainer),
+    OrgsToMigrate = [ db_results_to_org(OrgDbRow, Superuser) || OrgDbRow <- OrgsToMigrateResults],
+    mover_transient_migration_queue:initialize_queue(?MODULE, OrgsToMigrate).
+
+db_results_to_requestor(FieldsValues) ->
+    {<<"username">>, Name} = proplists:lookup(<<"username">>, FieldsValues),
+    {<<"id">>,  Id}        = proplists:lookup(<<"id">>, FieldsValues),
+    {<<"authz_id">>,  AzId} = proplists:lookup(<<"authz_id">>, FieldsValues),
+    #mover_requestor{username = Name, id = Id, authz_id = AzId}.
+
+db_results_to_org(FieldsValues, Superuser) ->
+    %% Each object will be a list like this:
+    %% [{<<"name">>,<<"ponyville">>},
+    %%  {<<"id">>,<<"1e9ed26e6cbcdeedb128e896aa0f34c2">>},
+    %%  {<<"authz_id">>,<<"8184b014cba9484eabafb8568d299e67">>}
+    {<<"name">>, Name}     = proplists:lookup(<<"name">>, FieldsValues),
+    {<<"id">>,  Id}        = proplists:lookup(<<"id">>, FieldsValues),
+    {<<"authz_id">>, AzId} = proplists:lookup(<<"authz_id">>, FieldsValues),
+    #mover_org{name = Name, id = Id, authz_id = AzId, superuser = Superuser}.
+
+migration_start_worker_args(Object, AcctInfo) ->
+    [Object, AcctInfo].
+
+next_object() ->
+    mover_transient_migration_queue:next(?MODULE).
+
+migration_action(Org, _AcctInfo) ->
+    try
+        upgrade_org(Org)
+    catch
+        exit:Wat ->
```

Comment:
With the capture of this in mover_transient_worker, is this still necessary? 

Once you add the callback for migration_complete(SuccessOrFailure), you can do the necessary shutdown there and not have to spread it in multiple places - you'd call it from mover_transient_queue_batch_migrator:migrate_all as the last thing.